app/actions/views/channel.js

```diff
@@ -393,31 +393,20 @@ export function handleSelectChannel(channelId, fromPushNotification = false) {
                 teamId: currentTeamId,
                 channelId,
             },
-        ];
-
-        let markPreviousChannelId;
-        if (!fromPushNotification && !sameChannel) {
-            markPreviousChannelId = currentChannelId;
-            actions.push({
-                type: ViewTypes.SELECT_CHANNEL_WITH_MEMBER,
-                data: currentChannelId,
-                channel: getChannel(state, currentChannelId),
-                member: getMyChannelMember(state, currentChannelId),
-            });
-        }
-
-        if (!fromPushNotification) {
-            actions.push({
+            {
                 type: ViewTypes.SELECT_CHANNEL_WITH_MEMBER,
                 data: channelId,
                 channel,
                 member,
-            });
-        }
+            },
+        ];
 
         dispatch(batchActions(actions));
 
-        dispatch(markChannelViewedAndRead(channelId, markPreviousChannelId));
+        if (!fromPushNotification && !sameChannel) {
+            const markPreviousChannelId = currentChannelId;
+            dispatch(markChannelViewedAndRead(channelId, markPreviousChannelId));
```

Comment:
this should be dispatched regardless, right? as you want to mark the channel as read